hudi-common/src/test/java/org/apache/hudi/common/table/read/TestCustomMerger.java

```diff
@@ -0,0 +1,197 @@
+    ClosableIterator<IndexedRecord> iterator = getFileGroupIterator(2);
+    List<String> leftKeysExpected =
+        Arrays.asList("6", "7", "8", "9", "10");
+    List<String> leftKeysActual = new ArrayList<>();
+    while (iterator.hasNext()) {
+      leftKeysActual.add(iterator.next()
+          .get(AVRO_SCHEMA.getField(ROW_KEY).pos())
+          .toString());
+    }
+    assertEquals(leftKeysExpected, leftKeysActual);
+  }
+
+  @Test
+  public void testWithTwoLogFiles() throws IOException, InterruptedException {
+    ClosableIterator<IndexedRecord> iterator = getFileGroupIterator(3);
+    List<String> leftKeysExpected =
+        Arrays.asList("1", "3", "6", "7", "8", "9", "10");
+    List<String> leftKeysActual = new ArrayList<>();
+    while (iterator.hasNext()) {
+      leftKeysActual.add(iterator.next()
+          .get(AVRO_SCHEMA.getField(ROW_KEY).pos())
+          .toString());
+    }
+    assertEquals(leftKeysExpected, leftKeysActual);
+  }
+
+  @Test
+  public void testWithThreeLogFiles() throws IOException, InterruptedException {
+    ClosableIterator<IndexedRecord> iterator = getFileGroupIterator(4);
+    List<String> leftKeysExpected =
+        Arrays.asList("1", "3", "7", "9", "10");
+    List<String> leftKeysActual = new ArrayList<>();
+    while (iterator.hasNext()) {
+      leftKeysActual.add(iterator.next()
+          .get(AVRO_SCHEMA.getField(ROW_KEY).pos())
+          .toString());
+    }
+    assertEquals(leftKeysExpected, leftKeysActual);
+  }
+
+  @Test
+  public void testWithFourLogFiles() throws IOException, InterruptedException {
+    ClosableIterator<IndexedRecord> iterator = getFileGroupIterator(5);
+    List<String> leftKeysExpected =
+        Arrays.asList("1", "3", "7", "9", "10");
+    List<String> leftKeysActual = new ArrayList<>();
+    while (iterator.hasNext()) {
+      leftKeysActual.add(iterator.next()
+          .get(AVRO_SCHEMA.getField(ROW_KEY).pos())
+          .toString());
+    }
+    assertEquals(leftKeysExpected, leftKeysActual);
+  }
+
+  public static class CustomAvroMerger implements HoodieRecordMerger {
+    public static final String KEEP_CERTAIN_TIMESTAMP_VALUE_ONLY =
+        "KEEP_CERTAIN_TIMESTAMP_VALUE_ONLY";
+    public static final String TIMESTAMP = "timestamp";
+
+    @Override
+    public Option<Pair<HoodieRecord, Schema>> merge(
+        HoodieRecord older,
+        Schema oldSchema,
+        HoodieRecord newer,
+        Schema newSchema,
+        TypedProperties props
+    ) throws IOException {
+      if (newer.getOrderingValue(newSchema, props).compareTo(
+          older.getOrderingValue(oldSchema, props)) >= 0) {
+        if (newer.isDelete(newSchema, props)) {
+          return Option.empty();
+        }
+        int id = Integer.parseInt(((HoodieAvroIndexedRecord) newer)
+            .getData().get(newSchema.getField(ROW_KEY).pos()).toString());
+        if (id % 2 == 1L) {
+          return Option.of(Pair.of(newer, newSchema));
+        }
```

Comment:
Keep the old record otherwise?  or the intent is to delete the older after merging?